include/odp/api/spec/traffic_mngr.h

```diff
@@ -331,6 +354,31 @@ typedef struct {
 	/** The per_level array specifies the TM system capabilities that
 	 * can vary based upon the tm_node level. */
 	odp_tm_level_capabilities_t per_level[ODP_TM_MAX_LEVELS];
+
+	/** Indicates the packet priority modes supported by a TM systems on a platform.
+	 * A platform can support multiple packet priority modes. The actual mode,
+	 * a TM system runs with is defined by odp_tm_requirements_t.
+	 */
+	odp_bool_t pkt_prio_modes[ODP_TM_PKT_PRIO_MODE_MAX];
+
+	/** max_wfq_groups_per_node indicates maximum number of wfq groups supported
+	 * by a tm node at any level. This capability is valid only when packet
+	 * priorirty mode is ODP_TM_PKT_PRIO_MODE_OVERRIDE.
+	 *
+	 * In ODP_TM_PKT_PRIO_MODE_OVERRIDE mode, since all pkts from a fan-in come
+	 * with same priority, a WFQ group is formed by multiple fan-in's with
+	 * same packet priority. This capability indicates that at a given node, at
+	 * max only so many wfq groups can have multiple fan-in's. Rest of
+	 * the WFQ groups can only have one fan-in. This also implies that rest of
+	 * the WFQ groups with single fan-in act as static priority fan-in.
+	 * In ODP_TM_PKT_PRIO_MODE_GLOBAL mode, since packet priority is global,
+	 * every node should support as many number of WFQ groups as number of
+	 * priorities available. Hence this field is not applicable.
+	 *
+	 * This value can range from 1..ODP_TM_MAX_PRIORITIES as there can be
+	 * only as many number of WFQ groups as number of priorities.
+	 */
+	uint8_t max_wfq_groups_per_node;
```

Comment:
I do not think there is a need for this new WFQ group concept. I suppose this could be simply max_num_schedulers_per_node.

> This capability indicates that at a given node, at max only so many wfq groups can have multiple fan-in's. Rest of the WFQ groups can only have one fan-in.

This is a bit inconsistent since max_wfq_groups does not here actually mean the maximum number of WFQ groups but the maximum number of WFQ groups that have more than one fan-in.

> every node should support as many number of WFQ groups as number of

This should be "...every node supports as many WFQ groups as the number of ..."